lib/lz4.c

```diff
@@ -193,10 +193,18 @@
  * and be available at link time */
 void* LZ4_malloc(size_t s);
 void* LZ4_calloc(size_t n, size_t s);
+#ifdef LZ4_FREE_ACCEPTS_SIZE
+void  LZ4_free(void* p, size_t s);
+#else
 void  LZ4_free(void* p);
+#endif
 # define ALLOC(s)          LZ4_malloc(s)
 # define ALLOC_AND_ZERO(s) LZ4_calloc(1,s)
+#ifdef LZ4_FREE_ACCEPTS_SIZE
+# define FREEMEM(p)        LZ4_free(p,sizeof(*(p)))
```

Comment:
For the "size" parameter to be always correct with just this simple trick,
it implies that `p`  has always a well defined type, e.g. not incomplete, not `void*`.
While this is generally the case, there are a few exceptions.

In `lz4hc.c`, there is an old  (deprecated but still supported) function `LZ4_freeHC()` that's still using  a `void*`.
It would need to be updated, to ensure it propagates the correct size.

More difficult, in case of library amalgamation, this new definition of `FREEMEM()` is propagated to `lz4frame.c`.
In this library, `FREEMEM()` is used a lot more, and there is a need to ensure that every time, the right "size" parameter is passed. I'm skeptical that `sizeof(*(p))` get it always right, as the use cases are more complex in this library.

Reply:
I guess "this one simple trick" only works for clickbait ads and not for code updates.  😁 I'll rework the solution.  Thank you for your feedback.